src/lib/io/include/sol-uart.h

```diff
@@ -238,7 +251,18 @@ void sol_uart_close(struct sol_uart *uart);
  * @note Caller should guarantee that tx buffer will not be freed until
  * callback is called.
  */
-bool sol_uart_write(struct sol_uart *uart, const uint8_t *tx, unsigned int length, void (*tx_cb)(void *data, struct sol_uart *uart, uint8_t *tx, int status), const void *data);
+int sol_uart_write(struct sol_uart *uart, const uint8_t *tx, size_t length, void (*tx_cb)(void *data, struct sol_uart *uart, uint8_t *tx, int status), const void *data);
```

Comment:
as per today's talk in person, I'm wondering if we should convert these to the same pattern, taking a blob and a single CB given to config.

@ceolin is doing the socket, at least trying to unify with that. IMO the blob is a good trade-off for memory consumption.